File: Python/EstruturaDeRepeticao/exercicio_12.py
"""
Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer número inteiro entre 1 a 10.
O usuário deve informar de qual numero ele deseja ver a tabuada.
A saída deve ser conforme o exemplo abaixo:

Tabuada de 5:
5 X 1 = 5
5 X 2 = 10
...
5 X 10 = 50
"""

numero = int(input(f"Digite o número para ver a tabuada (entre 1 e 10): "))
if numero > 10 or numero < 1:
    while numero > 10 or numero < 1:
        print("Por favor, digite um número entre 1 e 10.")
        numero = int(input(f"Digite o número para ver a tabuada (entre 1 e 10): "))

print(f"Tabuada do {numero}:")
for i in range(1, 11):
    resultado = numero * i
    print(f"{numero} x {i} = {resultado}")

File: Java/EstruturaDeRepeticao/Exercicio12.java
// Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer número inteiro entre 1 a 10.
// O usuário deve informar de qual numero ele deseja ver a tabuada.
// A saída deve ser conforme o exemplo abaixo:

// Tabuada de 5:
// 5 X 1 = 5
// 5 X 2 = 10
// ...
// 5 X 10 = 50
package EstruturaDeRepeticao;
import java.util.Scanner;

public class Exercicio12 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite um número inteiro entre 1 e 10 para ver a tabuada: ");
        int numero = scanner.nextInt();

        while (numero < 1 || numero > 10) {
            System.out.print("Número inválido! Digite um número inteiro entre 1 e 10: ");
            numero = scanner.nextInt();
        }

        System.out.println("Tabuada de " + numero + ":");

        for (int i = 1; i <= 10; i++) {
            int resultado = numero * i;
            System.out.println(numero + " x " + i + " = " + resultado);
        }

        scanner.close();
    }
}
